android/src/com/mapswithme/maps/review/ReviewAdapter.java

```diff
@@ -0,0 +1,226 @@
+package com.mapswithme.maps.review;
+
+import android.support.annotation.CallSuper;
+import android.support.v7.widget.RecyclerView;
+import android.text.TextUtils;
+import android.text.format.DateFormat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.mapswithme.maps.R;
+import com.mapswithme.maps.widget.recycler.RecyclerClickListener;
+import com.mapswithme.util.UiUtils;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+
+class ReviewAdapter extends RecyclerView.Adapter<ReviewAdapter.BaseViewHolder>
+{
+  private static final int MAX_COUNT = 15;
+  private static final int VIEW_TYPE_REVIEW = 0;
+  private static final int VIEW_TYPE_MORE = 1;
+  private static final int VIEW_TYPE_RATING = 2;
+
+  private final ArrayList<Review> mItems;
+  private final RecyclerClickListener mListener;
+  private final String mRating;
+  private final int mRatingBase;
+
+  ReviewAdapter(ArrayList<Review> images, RecyclerClickListener listener, String rating,
+                int ratingBase)
+  {
+    mItems = images;
+    mListener = listener;
+    mRating = rating;
+    mRatingBase = ratingBase;
+  }
+
+  @Override
+  public BaseViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
+  {
+    if (viewType == VIEW_TYPE_REVIEW)
+    {
+      return new ReviewHolder(LayoutInflater.from(parent.getContext())
+                                            .inflate(R.layout.item_comment, parent, false), mListener);
+    }
+
+    if (viewType == VIEW_TYPE_MORE)
+    {
+      return new MoreHolder(LayoutInflater.from(parent.getContext())
+                                          .inflate(R.layout.item_more_button, parent, false), mListener);
+    }
+
+    return new RatingHolder(LayoutInflater.from(parent.getContext())
+                                          .inflate(R.layout.item_rating, parent, false), mListener);
+  }
+
+  @Override
+  public void onBindViewHolder(BaseViewHolder holder, int position)
+  {
+    int positionNoHeader = position - 1;
+
+    if (position == 0)
+    {
+      ((RatingHolder) holder).bind(mRating, mRatingBase);
+    } else if (positionNoHeader < mItems.size())
+    {
+      holder.bind(mItems.get(positionNoHeader), positionNoHeader);
+    } else
+    {
+      holder.bind(null, positionNoHeader);
+    }
+  }
+
+  @Override
+  public int getItemCount()
+  {
+    if (mItems == null)
+    {
+      return 1;
```

Comment:
Could you please comment why 1 instead of 0, and why MAX_COUNT + 2 below instead of MAX_COUNT?